### DISCUSSION TITLE
Generating a resource with `create source git`

### DISCUSSION THREAD

**Discussion by rjschwei** [fluxcd/flux2#3547] [Q&A]
Am I correct in concluding that there is no way to generate the necessary references for flux unless one has Internet access?

The doc states

"""The create source git command generates a GitRepository resource and waits for it to sync."""

and there appears to be no command where I can simply point to an existing git repo that I have already cloned. Basically when I am on an airplane I am out of luck. That appears particularly bad given the by design disconnected approach of git, i.e. not network needed until I want to push something. As such I should be able to test in a disconnected way as well.

**Comment by hiddeco** [comment]
Possible using the `--export` flag if I am following you correctly:

```console
$ flux create source git foo --url https://github.com/hiddeco/foo.git --branch main --export
---
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: GitRepository
metadata:
  name: foo
  namespace: flux-system
spec:
  interval: 1m0s
  ref:
    branch: main
  url: https://github.com/hiddeco/foo.git
```

**Comment by danielloader** [comment]
It reads to me like the request is along the lines of wanting to be able to bootstrap the source controller with either a volume mount to a local repository for rapid testing and offline airgapped work with a local cluster, or the same idea but just a one off tarballing of the local directory path and then decompressed into the source controller storage mount.

**Reply by hiddeco** [reply]
Instead of going through this pain, you would then be much better of using a local or in-cluster registry, in combination with https://fluxcd.io/flux/components/source/ocirepositories/.

**Reply by hiddeco** [reply]
For more inspiration around this, see e.g. https://github.com/stefanprodan/flux-local-dev
